overlord/snapstate/handlers.go

```diff
@@ -3518,6 +3561,22 @@ func (m *SnapManager) doRemoveAliases(t *state.Task, _ *tomb.Tomb) error {
 	}
 	snapName := snapsup.InstanceName()
 
+	var removeReason removeAliasesReason
+	if err := t.Get("remove-reason", &removeReason); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	skip, err := shouldSkipRemoveAliases(st, removeReason, snapsup.Type)
+	if err != nil {
+		return err
+	}
+	if skip {
+		// skip removing aliases, setup-aliases will prune old aliases later.
+		snapst.AliasesPending = false
+		Set(st, snapName, snapst)
```

Comment:
we probably don't need this as it should be already like that when we start

Reply:
updated, thanks!